source/io/module_javaScriptInvoke.js

```diff
@@ -474,6 +426,10 @@
             });
         };
 
+        var shouldUpdateReturnValue = function (isInternalFunction, returnTypeRef) {
+            return !isInternalFunction || Module.typeHelpers.isJSObjectRefnum(returnTypeRef);
```

Comment:
I'm a little confused here. If the function is internal we should not update the return value.

For internal functions the user is passed a bunch of valueRefs. They should use those valueRefs to update the return value. They should call vireo.eggShell.writeJavaScriptRefNum and vireo.eggShell.readJavaScriptRefNum.

If anything we should throw an error if a return value is provided (not undefined) and say they should use the valueRef and the API functions to update the return value.

Reply:
I cleaned up this code and got rid of shouldUpdateReturnValue business.